### PR TITLE
feat(categories): add new monitoring endpoint to retrieve service categories

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

This adds a new monitoring endpoint to retrieve service categories from the realtime. No check is made on BBDO v3 as tags table is always created but might be empty. So the endpoint will retrieve nothing

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

Later on

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/php/Core/Infrastructure/RealTime/Repository/DbServiceCategoryFactoryTest.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Core\Infrastructure\RealTime\Repository;
+
+use Core\Infrastructure\RealTime\Repository\ServiceCategory\DbServiceCategoryFactory;
+
+it('DbServiceFactory creation test', function () {
```
Comment (adr-mo) [inline tests/php/Core/Infrastructure/RealTime/Repository/DbServiceCategoryFactoryTest.php:28]:
```suggestion
it('DbServiceCategoryFactory creation test', function () {
```

**Review by TamazC — APPROVED** [review]

File: src/Core/Infrastructure/RealTime/Repository/ServiceCategory/DbReadServiceCategoryRepository.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        $statement = $this->db->prepare(
+            $this->translateDbName('SELECT id, name FROM `:dbstg`.tags WHERE type = :type')
+        );
+
+        $statement->bindValue(':type', Tags::SERVICE_CATEGORY_TYPE_ID, \PDO::PARAM_INT);
+
+        $statement->execute();
+
+        while ($record = $statement->fetch(\PDO::FETCH_ASSOC)) {
+            yield DbServiceCategoryFactory::createFromRecord($record);
```
Comment (adr-mo) [inline src/Core/Infrastructure/RealTime/Repository/ServiceCategory/DbReadServiceCategoryRepository.php:59]: Go back to a standard array for now, and then we will discuss the use of yield functionality later on

File: src/Core/Application/RealTime/UseCase/FindServiceCategory/FindServiceCategory.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+     * @param FindServiceCategoryPresenterInterface $presenter
+     */
+    public function __invoke(FindServiceCategoryPresenterInterface $presenter): void
+    {
+        $this->info('Searching for service categories');
+
+        try {
+            $serviceCategories = $this->repository->findAll();
+        } catch (\Throwable $e) {
+            $presenter->setResponseStatus(new ErrorResponse($e->getMessage()));
```
Comment (callapa) [inline src/Core/Application/RealTime/UseCase/FindServiceCategory/FindServiceCategory.php:52]: please to log error

Comment (adr-mo) [reply src/Core/Application/RealTime/UseCase/FindServiceCategory/FindServiceCategory.php:53]: done

**Review by callapa — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**